src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,162 @@
+        Node<T> newElem = new Node<>(null, value, first);
+        if (first.item == null) {
+            last.prev = newElem;
+        } else {
+            first.prev = newElem;
+        }
+        first = newElem;
+        size++;
+    }
+
     @Override
     public void add(T value) {
-
+        if (first.item == null) {
+            addToHead(value);
+        } else {
+            addToTail(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            checkIndexException(index);
+        }
+        if (index == size) {
+            if (first.item == null) {
+                addToHead(value);
+            } else {
+                addToTail(value);
+            }
+        } else {
+            Node<T> result = first;
+            for (int i = 0; i < size; i++) {
+                if (index == i) {
+                    Node<T> newElem = new Node<>(result.prev, value, result.prev.next);
+                    result.prev.next = newElem;
+                    result.prev = newElem;
+                    size++;
+                    break;
+                }
+                result = result.next;
+            }
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexException(index);
+        Node<T> result = first;
+        for (int i = 0; i < index; i++) {
+            result = result.next;
+        }
+        return result.item;
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndexException(index);
+        Node<T> result = first;
+        for (int i = 0; i < size; i++) {
+            if (i == index) {
+                result.item = value;
+            }
+            result = result.next;
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexException(index);
+        Node<T> result = first;
+        for (int i = 0; i < size; i++) {
+            if (index == 0) {
+                result.next.prev = result.prev;
+                first = result.next;
+                break;
+            } else if (index == i && i < size - 1) {
+                result.prev.next = result.next;
+                result.next.prev = result.prev;
+                break;
+            } else if (index == i && i == size - 1) {
+                result.prev.next = result.next;
+                last = result.prev;
+                break;
+            }
+            result = result.next;
+        }
+        size--;
+        return result.item;
     }
 
     @Override
     public T remove(T t) {
+        Node<T> result = first;
+        for (int i = 0; i < size; i++) {
+            if (result.item == t) {
+                return remove(i);
+            }
+            result = result.next;
```

Comment:
@Korfat shouldn't you manage two way link here? You linked to the new node only in one direction